ports/xgboost/portfile.cmake

```
@@ -0,0 +1,40 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO dmlc/xgboost
+    REF v${VERSION}
+    SHA512 4465f383df70ee415faaeb745459bfc413f71bff0d02e59e67173975188bed911044dbea1a456550496f8c5a7c0b50002275b6be72c87386a6118485e1e41829
+    HEAD_REF master
+)
+
+# Set the expected path of the dmlc-core directory inside the xgboost source directory
+vcpkg_from_github(
+    OUT_SOURCE_PATH DMLC_CORE_SRC
+    REPO dmlc/dmlc-core
+    REF 81db539486ce6525b31b971545edffee2754aced
+    SHA512 9b288fd1ceeef0015e80b0296b0d4015238d4cc1b7c36ba840d3eabce87508e62ed5b4fe61504f569dadcc414882903211fadf54aa0e162a896b03d7ca05e975
+    HEAD_REF master
+)
+
+# Custom CMake script to move dmlc-core to the correct location
+file(REMOVE_RECURSE "${SOURCE_PATH}/dmlc-core")
+file(RENAME "${DMLC_CORE_SRC}" "${SOURCE_PATH}/dmlc-core")
+
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}
+)
+
+vcpkg_cmake_install()
+
+# Adjustments to address the warnings
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+vcpkg_fixup_pkgconfig()
+vcpkg_cmake_config_fixup(PACKAGE_NAME xgboost CONFIG_PATH "lib/cmake")
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
+endif()

```

Any executables to be handled?

---

@dg0yt  thank you for the review. Not sure if I understand, what executables you were referring to?  This code snippet  added based on the post-build check's warnings and its fix suggestions.

---

Given that the directory was created, I suspect that it was created to install either exectuables or DLLs.  
In the case of executables, this approach would create inconsistent behavior for static vs dynamic. `vcpk_copy_tools(... AUTO_CLEAN)` would give consistent behavior. 
In the case of DLLs, they must either be necessary, or they indicate insufficient control of library linkage.

---

@dg0yt the generated files under  "${CURRENT_PACKAGES_DIR}/bin"  and "${CURRENT_PACKAGES_DIR}/debug/bin" are two files: xgboost.exe and xgboost.dll.  
I replaced the above code with `vcpk_copy_tools` but I see the same warning and the same suggestion to remove the bin and debug/bin folders:
`vcpkg_copy_tools(
    TOOL_NAMES xgboost
    SEARCH_DIR "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin"
    AUTO_CLEAN
)`

`warning: If the creation of bin\ and/or debug\bin\ cannot be disabled, use this in the portfile to remove them
    if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
        file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
    endif()`

---

```suggestion
vcpkg_copy_tools(TOOL_NAMES xgboost AUTO_CLEAN)
```
must handles this, really.